firebase-firestore/src/androidTest/java/com/google/firebase/firestore/BundleTest.java

```diff
@@ -71,7 +72,7 @@
             + "{\"seconds\":1000,\"nanos\":9999},\"exists\":true}}",
         "{\"document\":{\"name\":\"projects/{projectId}/databases/(default)/documents/coll-1/b\","
             + "\"createTime\":{\"seconds\":1,\"nanos\":9},\"updateTime\":{\"seconds\":1,"
-            + "\"nanos\":9},\"fields\":{\"k\":{\"stringValue\":\"b\"},\"bar\":"
+            + "\"nanos\":9},\"fields\":{\"k\":{\"stringValue\":\"\uD83D\uDE0A\"},\"bar\":"
```

Comment:
Note that this test passes against `master` as well.

Reply:
Because the bundles built here also count chars, not bytes.

Reply:
See other change in this test, it uses byte count, which fails the tests.

Reply:
Sorry, I just saw this flaw in my logic. So the bug is that we try to read more data than available.